WcaOnRails/spec/controllers/api_competitions_controller_spec.rb

```diff
@@ -268,6 +278,27 @@
         parsed_body = JSON.parse(response.body)
         expect(parsed_body["id"]).to eq "TestComp2014"
       end
+
+      it 'gets wcif with consistent competitor_id' do
+        last_registration = nil
+        user_competitor_ids = []
+        comp_id = 1
+        3.times do
+          user = FactoryBot.create(:user)
+          user_competitor_ids << [user.id, comp_id]
+          comp_id += 1
+          last_registration = FactoryBot.create(:registration, :accepted, competition: competition, user: user)
+        end
+        get_wcif_and_compare_persons_to(competition.id, user_competitor_ids + [[competition.delegates.first.id, nil]])
+
+        # Move last registration to deleted
+        last_registration.deleted_at = Time.now
```

Comment:
Does this actually do anything? Don't you need to call `save!`? Or, to do it in one line:

```ruby
last_registration.update_attributes!(deleted_at: Time.now)
```

Reply:
Or even `last_registration.touch :deleted_at` =)

Reply:
Fixed, and tested the fail through a `shuffle` on registrations inserted in the WCIF generation :)